clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1467,6 +1474,27 @@ class KernelObjVisitor {
 public:
   KernelObjVisitor(SemaSYCL &S) : SemaSYCLRef(S) {}
 
+  static bool useTopLevelKernelObj(const CXXRecordDecl *KernelObj) {
+    // If the kernel is empty, "decompose" it so we don't generate arguments.
+    if (KernelObj->isEmpty())
+      return false;
+    // FIXME: Workaround to not change large number of tests
```

Comment:
I restricted the patch to nv for now as many (many) clang test need to be updated without this. We can force the flag to not decompose in all of them as well. But either way, I think we need proper addressing for that (I need to look on a case by case if we need or not the decomposition), so we should that in a follow up patch. I suspect this could trigger other issue for tracing and others, and that starts to be out of scope of what we need here.